src/js/plugins/tts/AbstractTTSEngine.js

```diff
@@ -0,0 +1,136 @@
+import AsyncStream from './AsyncStream.js';
+
+/**
+ * @export
+ * @typedef {Object} PageChunk
+ * @property {Number} leafIndex
+ * @property {String} text
+ * @property {DJVURect[]} lineRects
```

Comment:
I'm kind of curious about these three `typedef`s. What is the goal of these? Are they intended to define objects for external use? I'm inclined to define these as actual classes instead of defining them in a comment block like this, but I may not be understanding their use case. ie:
```
class PageChunk {
  constructor(leafIndex, text, lineRects) {
    this.leafIndex = leafIndex;
    this.text = text;
    this.lineRects = lineRects;
  }
}

class TTSEngineOptions {
  constructor(server, bookPath, onLoadingStart, ....) {
    this.server = server;
    this.bookPath = bookPath;
    ....
  }
}
```

Reply:
They're there to let us better document types, mainly (in these files or others). i.e. "What parameter does this function take? It takes a POJO with these fields". Other languages have features like `data class`es which would be appropriate for these, but I feel full `class`es might be a little too much, since these don't have any init logic or associated functionality (ie methods). They also cause a decent amount of duplication with the parameter name/`this.* = *`.

I think converting these to classes sets a precedent that _all_ POJOs (e.g. `{ x: 'foo' }`) should be converted to es6 classes, which I kind of disagree with.